customers.go

```diff
@@ -22,10 +25,30 @@ type Customer struct {
 	UpdatedAt time.Time
 }
 
+// Customers is a collection of customers
+type Customers []Customer
+
+// CustomerSearchQuery contains all the supported search queries
+type CustomerSearchQuery struct {
+	Tag string
+}
+
+func (c *CustomerSearchQuery) String() string {
+	params := url.Values{}
+
+	if c.Tag != "" {
+		params.Add("customer_tag:", c.Tag)
+	}
+
+	return params.Encode()
+}
+
 // CustomerRepository maintains the customers of a shop.
 type CustomerRepository interface {
 	// Update updates a single customer
 	Update(customer Customer) (Customer, error)
 	// Get gets customer with the provided id
 	Get(id int64) (Customer, error)
+	// GetByQuery gets the customers matching the query and returns the fields requested
+	GetByQuery(fields []string, query CustomerSearchQuery) (Customers, error)
```

Comment:
?
```suggestion
	Search(fields []string, query CustomerSearchQuery) (Customers, error)
```

Reply:
Think the current one is a little more self-explanatory 🤔. We say it's a GET and also indicates that we are getting the customer by a certain query.

In contrast Search is also good since more compact and you can see what needs to be passed in and what it returns.
No preference, I can change it if that sounds better 👍🏻

Reply:
I think I'd prefer search but not too fussed either way tbh. So to save time, lets keep it as it is :).

Reply:
I just updated the comment also as it sounded too implementation specific.